MTM-59277 Add RN for measurement series validation (#2497)
SoftwareAG/c8y-docs: content/change-logs/platform-services/cumulocity-10-20-591-0-fail-series-validation-more-than-1-dot.md

---
date: 

> title: Measurement series endpoint returns error if series contains more than one dot

Reviewer: ```suggestion
title: Measurement series endpoint returns an error if the series contains more than one dot
```

product_area: Platform services
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-OG_650_b2
    label: Core platform
build_artifact:
  - value: tc-QHwMfWtBk7
    label: cumulocity
ticket: MTM-59277
version: 10.20.591.0
---

> Property names used for fragment and series must not contain whitespaces nor the special characters (https://cumulocity.com/api/core/#operation/postMeasurementCollectionResource)

Reviewer: ```suggestion
Property names used for fragment and series must not contain whitespaces nor special characters (https://cumulocity.com/api/core/#operation/postMeasurementCollectionResource).
```

> Previously, `/measurement/measurements/series` endpoint accepted `series` query parameter which contains more than one dot, althought it returns empty response.

Reviewer: ```suggestion
Previously, the `/measurement/measurements/series` endpoint accepted a `series` query parameter which contained more than one dot, although it returned an empty response.
```

> With this change the endpoint returns `422` error if `series` query parameter contains more than one dot.

Reviewer 2: I think you should mention that dots that are not fragment and series separator should be escaped. Right ?

Author: That's good question, do we want to admit that it's supported (when escaped) as an opposite to what https://cumulocity.com/api/core/#operation/postMeasurementCollectionResource says? 
Maybe we should document it here https://cumulocity.com/api/core/#operation/getMeasurementSeriesResource not in RN?

Author: we discussed it and agreed not to share this information publicly

> With this change the endpoint returns `422` error if `series` query parameter contains more than one dot.

Reviewer: ```suggestion
With this change the endpoint returns a `422` error if the `series` query parameter contains more than one dot.
```
